### PR TITLE
debian: add extra checks when debian/snapd.postrm purge is run

### PR DESCRIPTION
**Pull request by mvo5**
This should addresses the concerns in https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1605303/comments/4

### CODE REVIEW

**Comment by niemeyer** [comment]
Looks fine, but the test failures seem related.

**Comment by mvo5** [comment]
Sorry that this took so long, test failure(s) fixed, new test added and code improved.

File: debian/snapd.postrm [hunk 1]
```diff
@@ -4,30 +4,54 @@ set -e
...
+            continue
+        fi
+
         echo "Stopping $unit"
         systemctl_stop $unit
 
-        echo "Removing $mount"
+        # if it is a mount unit, we can find the snap name in the mount
+        # unit (we just ignore unit files)
+        snap="$(grep 'Where=/snap/' /etc/systemd/system/$unit|cut -f3 -d/)"
```
Comment (chipaca) [inline debian/snapd.postrm]: no quotes around `$unit`? (and the quotes around `$()` re not needed AFAIK)

**Comment by chipaca** [comment]
:+1:

File: debian/snapd.postrm (hunk 1 above)
Comment (mvo5) [inline debian/snapd.postrm]: Excellent point(s), thank you!

**Pull request merged**